cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java

```diff
@@ -48,43 +51,54 @@
 @Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/configuration/dashboards")
 public class DashboardHttpHandler extends AuthenticatedHttpHandler {
   private static final Logger LOG = LoggerFactory.getLogger(DashboardHttpHandler.class);
+  private static final String CONFIG_TYPE = "dashboard";
+  private static final String PROPERTY_NAME = "config";
+  private static final String ID = "id";
 
-  //TODO: https://issues.cask.co/browse/CDAP-699 PersistenceStore will be used instead of inMemory implementation.
-  private final Table<String, String, String> configStore;
+  private final ConfigStore configStore;
 
   @Inject
-  public DashboardHttpHandler(Authenticator authenticator) {
+  public DashboardHttpHandler(Authenticator authenticator, ConfigStore configStore) {
     super(authenticator);
-    this.configStore = HashBasedTable.create();
+    this.configStore = configStore;
   }
 
   @Path("/")
   @POST
-  public synchronized void create(final HttpRequest request, final HttpResponder responder,
+  public void create(final HttpRequest request, final HttpResponder responder,
                                   @PathParam("namespace-id") String namespace) throws Exception {
     String data = request.getContent().toString(Charsets.UTF_8);
-    if (!data.equals("") && !isValidJSON(data)) {
+    // Initialize with empty config if no data is sent during creation of the dashboard
+    if (data.equals("")) {
```

Comment:
A dashboard can be created without any configuration. So that will be a POST with no body in which case data = "". So in that case I thought I will create a empty JSON. we can choose not to have a empty Json "{}" if that is not required.